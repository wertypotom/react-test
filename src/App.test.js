import React from 'react';
import renderer from 'react-test-renderer';
import App, { Counter, dataReducer } from './App';

const list = ['a', 'b', 'c']

describe('Counter', () => {
    test('snapshot tests', () => {
        const component = renderer.create(<Counter counter={1} />)
        let tree = component.toJSON()

        expect(tree).toMatchSnapshot()

    })
})

describe('App', () => {
    test('snapshot tests', () => {
        const component = renderer.create(<App />)
        let tree = component.toJSON()

        expect(tree).toMatchSnapshot()
    })

    describe('Reducer', () => {
        test('should set a list', () => {
            const state = { list: [], error: null }
            const newState = dataReducer(state, {
                type: 'SET_LIST',
                list
            })

            expect(newState).toEqual({ list, error: null })
        })

        test('should reset the error', () => {
            const state = { list: [], error: true }
            const newState = dataReducer(state, {
                type: 'SET_LIST',
                list
            })

            expect(newState).toEqual({ list, error: null })
        })

        test('should set the error', () => {
            const state = { list: list, error: true }
            const newState = dataReducer(state, {
                type: 'SET_ERROR',
            })

            expect(newState.error).toBe(true)
            expect(newState).toEqual({ list: [], error: true })
        })

    })
})