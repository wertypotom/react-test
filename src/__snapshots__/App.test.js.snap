// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot tests 1`] = `
<div>
  <h2>
    My Counter
  </h2>
  <div>
    <p>
      0
    </p>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
  <h2>
    My Async Data
  </h2>
  <ul />
</div>
`;

exports[`Counter snapshot tests 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;
